src/cr/cube/cubepart.py

```diff
@@ -844,7 +844,20 @@ def table_base(self):
 
     @lazyproperty
     def table_base_unpruned(self):
-        return self._matrix.table_base_unpruned
+        """np.int64 scalar or a 1D or 2D ndarray of np.int64 representing table base.
+
+        This value includes hidden vectors, those with either a hide transform on
+        their element or that have been pruned (because their base (N) is zero). This
+        does not affect a scalar value but when the return value is an ndarray, the
+        shape may be different than the array returned by `.table_base`.
+
+        A multiple-response (MR) dimension produces an array of table-base values
+        because each element (subvariable) of the dimension represents a logically
+        distinct question which may not have been asked of all respondents. When both
+        dimensions are MR, the return value is a 2D ndarray. A CAT_X_CAT slice produces
+        a scalar value for this property.
+        """
+        return self._assembler.table_base_unpruned
```

Comment:
This pattern keeps repeating itself, where we only give back a single prop from the `_assembler`. It feels a little fishy, and as if it's supposed to go away. Do you have something like that envisioned in the future iterations?

Reply:
No, actually. This is the intended pattern for measures and marginals (and table totals and labels). All that work is completely delegated to the `Assembler` object (which probably needs a slightly different name, since "assembling" describes _how_ it does it, not what it provides). Note that not all _Slice methods are measures and marginals, there are quite a few dimension characteristics for example, and something like `.column_percentages` probably doesn't need to come from assembler. So this pattern is evident on _many_ `_Slice` properties, but not all of them. The role of `_Slice` is to operate as the interface object, integrating the efforts of several collaborators to provide a unified public interface to the caller. We could not just collapse `Assembler` into `_Slice`, for example.

One reason all the measure and marginal computation needs to be in (and below) `Assembler` is that it needs to be accessible to internal objects, like for smoothing, sort-by-value, pairwise comparison, etc. Previously some of those extended the `_Slice` public interface for private purposes, which is not a good thing. The only one accessing `_Slice` properties should be the client.

This pattern will actually become more prevalent with the measure consolidation work.